### ISSUE TITLE
Cycle detection in validator does not work if involved reactors are imported

### ISSUE THREAD

**Issue by lhstrh** [lf-lang/lingua-franca#446]
Consider this minimal example:
* `Bar.lf`:
```
target C;

import Foo from "Foo.lf"
import Baz from "Baz.lf"

main reactor {
    foo = new Foo();
    baz = new Baz();
    foo.y -> baz.x;
    baz.y -> foo.x;
}
```
* `Foo.lf`:
```
target C;

reactor Foo {
    input x:int;
    input z:int;
    output y:int;
    
    reaction (x) {=
    =}
    
    reaction (z) -> y {=
    =}
}
```
* `Baz.lf`:
```
target C;

reactor Baz {
    input x:int;
    output y:int;
	
    reaction (x) -> y {= 
    =}
}

```
The cycle is not caught during validation. Instead (for the C target), the cycle is detected in the code generator when levels are assigned.

If only one of the reactors is imported (and the other defined in the same file as the main reactor), the problem is not present.

**Comment by edwardalee** [comment]
The offending code for this appears to be here in the validator: 
```
    @Check(FAST)
    def checkConnection(Connection connection) {

        // Report if connection is part of a cycle.
        for (cycle : this.info.topologyGraph.cycles) {
            for (lp : connection.leftPorts) {
                for (rp : connection.rightPorts) {
                    var leftInCycle = false
                    val reactorName = (connection.eContainer as Reactor).name

                    if ((lp.container === null && cycle.exists [
                        it.definition === lp.variable
                    ]) || cycle.exists [
                        (it.definition === lp.variable && it.parent === lp.container)
                    ]) {
                        leftInCycle = true
                    }

                    if ((rp.container === null && cycle.exists [
                        it.definition === rp.variable
                    ]) || cycle.exists [
                        (it.definition === rp.variable && it.parent === rp.container)
                    ]) {
                        if (leftInCycle) {
                            // Only report of _both_ reference ports are in the cycle.
                            error('''Connection in reactor «reactorName» creates ''' +
                                    '''a cyclic dependency between «lp.toText» and ''' +
                                    '''«rp.toText».''', Literals.CONNECTION__DELAY
                            )
                        }
                    }
                }
            }
        }
...
```
I have trouble reading this code, but I suspect that `it.definition` or `it.parent` is referring to the import statement rather than the reactor definition.

**Comment by lhstrh** [comment]
Almost. The issue is that the definition is in another file, which gets
ignored because the validator is supposed to report only errors in the file
that is being validated (the one that is being compiled). I will push a fix
soon.
-- 

--
Marten Lohstroh | Postdoctoral Researcher
University of California | 545Q Cory Hall
Berkeley, CA 94720  | +1 510 282 9135

**Comment by edwardalee** [comment]
Closed via PR #894.
